src/bootstrap/index.ts

```diff
@@ -77,13 +81,13 @@ async function run() {
         await exec.exec("sudo --preserve-env=http_proxy,https_proxy,no_proxy pip3 install tox");
         core.endGroup();
         core.startGroup("Install Juju");
-        await exec.exec("sudo snap install juju --classic");
+        await exec.exec("sudo snap install juju --classic --channel=${juju_channel}");
```

Comment:
You have to use backtick quotes for interpolation to work (this also applies to the other lines).

```suggestion
        await exec.exec(`sudo snap install juju --classic --channel=${juju_channel}`);
```